### PR TITLE
fix #43 Smarter handling of run and debug commands

### LINKED ISSUE #43: Improve debugEditorContents and runEditorContents command examples
Proposed improvements to the features recently added in support of https://github.com/microsoft/vscode/issues/92269

- Prefix the commands with `Mock Debug: `
- Hide the commands from the Command Palette
- Disable the commands when in debug mode, otherwise the editor command buttons remain active and could be mistaken for ways to interact with the debugging just launched. If they remain active and are clicked the user ends up with multiple concurrent debug sessions, which could be another source of confusion.

I will submit a PR shortly.

### PR DESCRIPTION
**Pull request by gjsjohnmurray**
This PR fixes #43 

The debug and run commands are now assigned to "Mock Debug" group so they get prefixed in Command Palette.

They only appear there if there is a suitable active document.

They can be run from the Command Palette or from a user-specified keybinding.

Their editor command buttons get disabled when debugging is active.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -18,26 +18,38 @@ const runMode: 'external' | 'server' | 'inline' = 'inline';
...
-				{
+		vscode.commands.registerCommand('extension.mock-debug.runEditorContents', (resource: vscode.Uri | undefined) => {
+			let targetResource = resource;
+			if (!targetResource && vscode.window.activeTextEditor) {
+				targetResource = vscode.window.activeTextEditor.document.uri;
+			}
+			if (targetResource) {
+				vscode.debug.startDebugging(undefined, {
+					type: 'mock',
+					name: 'Run Editor Contents',
```
Comment (isidorn) [inline src/extension.ts:29]: `Run File` is a better name. Or `Run Markdown File`

File: src/extension.ts [hunk 2]
```diff
@@ -18,26 +18,38 @@ const runMode: 'external' | 'server' | 'inline' = 'inline';
...
-			});
+		vscode.commands.registerCommand('extension.mock-debug.debugEditorContents', (resource: vscode.Uri | undefined) => {
+			let targetResource = resource;
+			if (!targetResource && vscode.window.activeTextEditor) {
+				targetResource = vscode.window.activeTextEditor.document.uri;
+			}
+			if (targetResource) {
+				vscode.debug.startDebugging(undefined, {
+					type: 'mock',
+					name: 'Debug Editor Contents',
```
Comment (isidorn) [inline src/extension.ts:48]: `Debug File` or `Debug Markdown File` is a better name

File: package.json [hunk 3]
```diff
@@ -77,17 +77,31 @@
...
+					"when": "resourceLangId == markdown"
+				}
 			]
 		},
 		"commands": [
 			{
 				"command": "extension.mock-debug.debugEditorContents",
 				"title": "Debug File",
+				"category": "Mock Debug",
+				"enablement": "!inDebugMode",
```
Comment (isidorn) [inline package.json:97]: Really like this! So that user would not accidently run it two times.

**Comment by isidorn** [comment]
Thanks for the PR, I really like it, but I leave this up to @weinand to do a final review.

File: src/extension.ts (hunk 2 above)
Comment (gjsjohnmurray) [reply src/extension.ts]: Done

File: src/extension.ts (hunk 1 above)
Comment (gjsjohnmurray) [reply src/extension.ts]: Done

**Comment by weinand** [comment]
@gjsjohnmurray thanks a lot for the PR.

**Comment by gjsjohnmurray** [comment]
> 
> @gjsjohnmurray thanks a lot for the PR.

@weinand you closed it unmerged. Was that intentional?

**Comment by weinand** [comment]
Yes, the structure of Mock Debug has completely changed and code has moved to new files. So it was easier to apply the changes from your PR manually in the new places.

**Pull request closed without merging**